xray/src/utils.rs

```diff
@@ -1,9 +1,24 @@
 use image::{GenericImage, GenericImageView, ImageResult, Pixel, RgbaImage, SubImage};
-use quadtree::NodeId;
+use quadtree::{NodeId, NODE_PREFIX};
+use std::io;
 use std::path::{Path, PathBuf};
 
+const META_PREFIX: &str = "meta";
+
 pub fn get_meta_pb_path(directory: &Path, id: NodeId) -> PathBuf {
-    directory.join(&format!("{}.pb", id).replace("r", "meta"))
+    directory.join(&format!("{}.pb", id).replace(&NODE_PREFIX.to_string(), META_PREFIX))
+}
+
+pub fn get_root_node_id_from_meta_pb_path(meta_path: &Path) -> io::Result<NodeId> {
+    let error_message = format!("Invalid path {:?}.", meta_path);
```

Comment:
How about changing this into
```rust
let invalid_input_error = || {
    io::Error::new(io::ErrorKind::InvalidInput, format!("Invalid path {:?}.", meta_path))
};
```
and calling `invalid_input_error()` below twice?

Reply:
Good idea! changed.